### ISSUE TITLE
bug: Shaders don't load when SMAA T1x option is renamed in engine_settings.xml (Enhanced Graphics mod)

### ISSUE THREAD

**Issue by 54ac** [aliasIsolation/aliasIsolation#59]
Using the [Enhanced Graphics (Plus) mod](https://www.nexusmods.com/alienisolation/mods/67) prohibits the shaders from loading due to the X being capitalized in the option for SMAA antialiasing in engine_settings.xml (SMAA T1X). If the file is edited manually to change it back to a lowercase x in line 196, the shaders work fine.

**Comment by RyanJGray** [comment]
Thanks for narrowing down the issue, I had noticed this myself!

I had a quick look through the game's code in Ghidra, and it doesn't appear to be an issue with Alias itself.

The reason this breaks Alias is because the engine code that checks those graphics settings **requires** the quality name to be `SMAA T1x` exactly, capitalising it (or changing the name in any other way) causes the engine code to disable anti-aliasing entirely, as it doesn't recognise what anti-aliasing method that option corresponds to :/

![image](https://github.com/user-attachments/assets/088b22b6-6cf8-4b6d-9f67-3c84dea7f066)

The TL;DR of that screenshot is:
If the quality name is neither `FXAA`, `SMAA T1x` or `SMAA T2x` _exactly_, then anti-aliasing is disabled

**Comment by 54ac** [comment]
Interesting, I would have assumed it checks the int value! Thanks for looking into it.

In any case, if anyone else stumbles upon this issue, use the [older version of the mod (non-Plus)](https://www.nexusmods.com/alienisolation/mods/34), it offers almost the same functionality without breaking antialiasing.

**Comment by RyanJGray** [comment]
No worries, thanks for doing the troubleshooting to work it out, I'm not a huge fan of the Plus version to be honest.

I would've assumed the same too, but I guess they decided the name strings made more sense for some reason.
